whereby/browser-sdk
# Clean up room connection status

### Summary

Clean up SDK's api surface before the beta release.

### Changes

* Remove `accepted` state from `roomConnectionStatus`
      > This state is not used for anything, it should be safe to remove.
* Rename `rejected` state to `knock_rejected`
      > Use a more descriptive status name when a knock request gets rejected on a locked room.
* Rename `roomConnectionStatus` -> `connectionStatus`
* Remove `mostRecentChatMessage`
* Remove `startScreenshareError`
* Replace `isStartingScreenshare` with `localScreenshareStatus` and fix the missing state reset when screen share stops
* Make roomConnectionOptions optional
       > Use default localMediaConstraints' audio, video: true.
* Use the default media constraints in the related story
       > `useLocalMedia` already has a default option and the same value was
passed in the story. This commit removes this unnecessary param.
* Do not expose `newJoiner` and `streams` on `remoteParticipant`
       > These are internal props, no need to expose them to in the useRoomConnection's `state`.

### Test plan

knock:
1. Use the sdk with a locked room
2. Join the room with a host key
3. Knock the room with a guest participant
4. Reject the knock with the host => room connection status should be `knock_rejected`

screenshare status:
1. start/stop screenshares and inspect the changes of the state's `localScreenshareStatus`
2. initially it is `undefined`
3. starting a screenshare => `starting`
4. when screenshare is in progress => `active`
5. stop screenshare either by the `stopScreenshare` action or by the browser's native alert => `undefined` again

useRoomConnection's default options:
1. use the sdk's `useRoomConnection` with a roomUrl only, like `useRoomConnection(roomUrl)`
2. it should connect with the default media constraints

`newJoiner` and `streams` props in `remoteParticipant`
1. `console.log` the `state` or `state.remoteParticipants` in your demo app
2. join the room with 2 participants
3. check the logs: `newJoiner` and `streams` props shouldn't be there.

Make sure your demo app works after migrating to this version 😇 .
